### PR TITLE
fly: 6.0.0 -> 6.1.0

### PR DESCRIPTION
**Pull request by flokli**
###### Motivation for this change
This bumps fly to the latest release, containing various fixes as described in https://github.com/concourse/concourse/releases/tag/v6.1.0.

We also stop shipping the bash completions in the nixpkgs tree and generate them during the build, as https://github.com/concourse/concourse/pull/4012 was merged.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/continuous-integration/fly/default.nix [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
   src = fetchFromGitHub {
     owner = "concourse";
     repo = "concourse";
     rev = "v${version}";
-    sha256 = "0chavwymyh5kv4fkvdjvf3p5jjx4yn9aavq66333xnsl5pn7q9dq";
+    sha256 = "14sm3xwhm6pfln18i9f9dyj7s2wcri43rxj4s1cja7nwqr5sqb3x";
   };
 
-  vendorSha256 = "127mc1wzqhn0l4ni6qxcx06qfdb1cgahzypjrs4vgr6i4sipjxck";
+  vendorSha256 = "0000000000000000000000000000000000000000000000000000";
```
Comment (aepsil0n) [inline pkgs/development/tools/continuous-integration/fly/default.nix:14]: This doesn't work. If I use `1c099sn5rrvj805va1lyjlbv7i2g1z5bxyaisv5l9365z0lv1cwm`, the package compiles fine though.

**Review by milibopp — CHANGES_REQUESTED** [review]
##### Reviewed points

- [x] package name fits guidelines
- [x] package version fits guidelines
- [ ] package build on x86_64, NixOS **fails**
- [ ] executables tested on x86_64, NixOS **fails**
- [ ] all depending packages build

##### Possible improvements

##### Comments

As commented in the code the `vendorSha256` hash is all zeros, which has to be changed to the actual hash of the vendored dependencies.

File: pkgs/development/tools/continuous-integration/fly/default.nix (hunk 1 above)
Comment (flokli) [reply pkgs/development/tools/continuous-integration/fly/default.nix]: erp, sorry, forgot to push that chunk. Updated.

**Review by milibopp — APPROVED** [review]
##### Reviewed points

- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64, NixOS
- [x] executables tested on x86_64, NixOS
- [x] all depending packages build

##### Possible improvements

none

##### Comments

Now it's all good! :)

I checked execution of `fly` on its own and the completion functionality specifically both in `zsh` and `bash` like that:

```zsh
source <(fly completion --shell zsh)
fly <TAB>
```

and I get a nice list of commands as completions.

**Pull request merged**